### PR TITLE
Fixed issue #15810 and added test for #5547

### LINKED ISSUE #15810: integrate(1/(2**(2*t/3)+1), (t,0,oo)) is wrong
`integrate(1/(2**(2*x/3)+1), (x,0,oo))` returns 1/2 while [it should be 3/2](https://www.wolframalpha.com/input/?i=Integrate%5B1%2F(2**(2*x%2F3)%2B1),+0,+inf%5D). The indefinite integral is also wrong.

Tested against 1.3 and master.

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Fixes #15810
Closes #5547 (added tests, fixed elsewhere)

#### Brief description of what is fixed or changed
For constant gcds, the result was wrong.

#### Release Notes

* integrals
   * Minor fix in case of constant gcd in Risch.

### CODE REVIEW

File: sympy/integrals/risch.py [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ def hermite_reduce(a, d, DE):
         ga, gd = ga.cancel(gd, include=True)
         dm = dm2
 
-    d = ds
+        d = ds
```
Comment (smichr) [inline sympy/integrals/risch.py:1112]: Was this intentional? neither `d` nor `ds` is updated in the loop. A better change might be to change 1090 to `d, r = d.div(dm)` and change `ds` in the loop to `d`.  But...again, this didn't fix anything, did it? Perhaps it just confirms that the issue is fixed (or else is intermittant).

Comment (oscargus) [reply sympy/integrals/risch.py:1112]: if `dm.degree(DE.t)` is 0 to start with you get a different value for `d`. So, yes, it actually did change the behaviour.

Comment (oscargus) [reply sympy/integrals/risch.py:1112]: But you are correct that one can write it in more clear way. I'll update.

**Comment by smichr** [comment]
How about this:
```python
    ga = Poly(0, DE.t)
    gd = Poly(1, DE.t)
    gagd = (ga, gd)

    dd = derivation(d, DE)
    dm = gcd(d, dd).as_poly(DE.t)
    d, _ = d.div(dm)

    while dm.degree(DE.t)>0:

        ddm = derivation(dm, DE)
        dm2 = gcd(dm, ddm)
        dms, _ = dm.div(dm2)
        d_ddm = d.mul(ddm)
        d_ddm_dm, _ = d_ddm.div(dm)

        b, c = gcdex_diophantine(-d_ddm_dm.as_poly(DE.t), dms.as_poly(DE.t), a.as_poly(DE.t))
        b, c = b.as_poly(DE.t), c.as_poly(DE.t)

        db = derivation(b, DE).as_poly(DE.t)
        d_dms, _ = d.div(dms)
        a = c.as_poly(DE.t) - db.mul(d_dms).as_poly(DE.t)

        ga = ga*dm + b*gd
        gd = gd*dm
        ga, gd = ga.cancel(gd, include=True)
        dm = dm2

    q, r = a.div(d)
    if (ga, gd) != gagd
        ga, gd = ga.cancel(gd, include=True)

    r, d = r.cancel(d, include=True)
    rra = q*fs[1] + fp*fs[1] + fs[0]
    rrd = fs[1]
    rra, rrd = rra.cancel(rrd, include=True)

    return ((ga, gd), (r, d), (rra, rrd))
```

**Comment by oscargus** [comment]
To be honest, I have no idea. I just happened to pick up how this could be solved, but I have not tried to understand the underlying math. I can give it a try over the weekend (if not earlier).

**Comment by jksuom** [comment]
`risch` is designed to work with polynomials over fields. Then the `gcd` of two coprime polynomials will automatically be 1. Ideally, (most) polynomials in `risch` should be created with `field=True`. As a provisional work-around, we can locally change the coefficients as follows:
```
--- a/sympy/integrals/risch.py
+++ b/sympy/integrals/risch.py
@@ -1086,13 +1086,13 @@ def hermite_reduce(a, d, DE):
     gd = Poly(1, DE.t)
 
     dd = derivation(d, DE)
-    dm = gcd(d, dd).as_poly(DE.t)
+    dm = gcd(d.to_field(), dd.to_field()).as_poly(DE.t)
     ds, r = d.div(dm)
 
     while dm.degree(DE.t)>0:
 
         ddm = derivation(dm, DE)
-        dm2 = gcd(dm, ddm)
+        dm2 = gcd(dm.to_field(), ddm.to_field())
         dms, r = dm.div(dm2)
         ds_ddm = ds.mul(ddm)
         ds_ddm_dm, r = ds_ddm.div(dm)
```

**Pull request closed without merging**